enclave-runtime/src/rpc/author/atomic_container.rs

```diff
@@ -0,0 +1,91 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use std::{
+	default::Default,
+	sync::{
+		atomic::{AtomicPtr, Ordering},
+		Arc, SgxMutex,
+	},
+};
+
+/// Generic atomic container that holds an item in a container
+pub struct AtomicContainer {
+	atomic_ptr: AtomicPtr<()>,
+}
```

Comment:
This part of the code used to be in the `top_pool_container.rs` file. Decided to extract it to a separate file now, since it might be re-used for various applications.